server/src/main/java/p2/service/impl/ProductService.java

```diff
@@ -0,0 +1,39 @@
+package p2.service.impl;
+import java.util.List;
+import p2.dao.impl.ProductDAO;
+import p2.model.Product;
+
+
+public class ProductService{
```

Comment:
It is imo cleaner to have a single instance variable called productDAO than have to instantiate a new object in every method

Reply:
Also no need to override GenericDAO methods in child classes.